### ISSUE TITLE
OnMoveItem Issue

### ISSUE THREAD

**Issue by PrinterLUA** [otland/forgottenserver#2202]
### Before creating an issue, please ensure:
- [x ] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [ x] This issue is reproducible without changes to the code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
Just move an item and check if it prints.
`function Player:onMoveItem(item, count, fromPosition, toPosition, fromCylinder, toCylinder)
	print("Moving Item")
	return true
end`

### Expected behaviour
It should not print anything if the item failes to get moved. Such as if a container is full and you try move item inside. Ofcouse it does not move, but the event still prints like it got moved.

### Actual behaviour
It should only print if the item successfully got moved from point A to point B

### Environment
TFS 1.2

**Comment by djarek** [comment]
Isn't this somewhat related to [this](https://github.com/otland/forgottenserver/issues/278)? The lua function is called once when querying the possibility of the move and then the second time when it is actually moved?

**Comment by PrinterLUA** [comment]
No, the thing is onMoveItem should only print if i was succed with the movement. So if i try to move the item to a Wall and ofc it does not move. Its not possible. But the event is printing like it was successfully moved.

**Comment by PrinterLUA** [comment]
Also while we are at it. There is another issue:https://github.com/otland/forgottenserver/issues/1804

**Comment by conde2** [comment]
I don't think this is a Bug because the player actually does try to move the item firing the event, but the movement is not successful, but even thought IMHO the event should still be called, Just the way it is.

**Comment by djarek** [comment]
I'm going to label this one as an enhancement, since this seems to be unexpected behavior rather than a bug,

**Comment by PrinterLUA** [comment]
Maybe add a check if the movement was successful or not?

**Comment by JSkalskiSBG** [comment]
@PrinterLUA 
**This can be closed.** Merged 7 months ago in #2474 
https://github.com/kito2/mtibia_master/commit/c2d5f1f60f233bb24324dd00028a4de1f53d4808
```
function Player:onMoveItem(item, count, fromPosition, toPosition, fromCylinder, toCylinder) 
  print("Moving Item")
  return true
end
```
will still print "Moving Item" as role of that event is to block item movement by LUA. You cannot block item movement AFTER it's moved, so it's called before 'internalMoveItem' (which can fail).

There is new event:
```
function onItemMoved(self, item, count, fromPosition, toPosition, fromCylinder, toCylinder)
  print("MovED Item")
end
```
which is executed after successful item move. Use **onItemMoved** instead of **onMoveItem**, if you want to print some notification.
